File: frontend/src/types/triage.ts
export type TriageScenario =
  | "housing"
  | "labor"
  | "consumer"
  | "debt"
  | "ip_reputation"
  | "other";

export interface EvidenceSlots {
  counterparty: string | null;
  time_place: string | null;
  amount: string | null;
  agreement: string | null;
  breach_fact: string | null;
  existing_evidence: string[];
}

export interface SimilarCase {
  title: string;
  summary: string;
  judgment: string;
  takeaway: string;
}

export interface TriageSidebar {
  evidence_slots: EvidenceSlots;
  action_steps: string[];
  demand_letter: string | null;
  similar_cases: SimilarCase[];
}

export interface ConversationMessage {
  role: "user" | "assistant";
  content: string;
  thinkingSteps?: { stage: string; content: string }[];
}

export interface AssistantTurn {
  role: "assistant";
  content: string;
  stage: string;
}

export interface StageProgress {
  current_stage: string;
  completed_stages: string[];
  remaining_stages: string[];
  next_stage: string | null;
  collected_slots: string[];
  missing_slots: string[];
  evidence_completion_ratio: number;
  evidence_complete: boolean;
  deliverables_ready: boolean;
}

export interface FrontendGuidance {
  display_mode: "slot_collection" | "action_guidance" | "deliverables_ready";
  primary_panel: "evidence_slots" | "action_steps" | "demand_letter";
  input_placeholder: string;
  follow_up_questions: string[];
  suggested_actions: string[];
}

export interface TriageResponse {
  session_id: string;
  scenario: TriageScenario;
  scenario_label: string;
  assistant_message: string;
  current_stage: string;
  assistant_turn: AssistantTurn;
  conversation_history: ConversationMessage[];
  progress: StageProgress;
  frontend: FrontendGuidance;
  sidebar: TriageSidebar;
}

export interface TriageRequest {
  user_input: string;
  scenario?: TriageScenario | null;
  scenario_hint?: TriageScenario | null;
  session_id?: string | null;
}

export type SSEEventType =
  | "session"
  | "stage_started"
  | "stage_completed"
  | "complete"
  | "error";

export interface SSESessionPayload {
  session_id: string;
  scenario: TriageScenario;
}

export interface SSEStageStartedPayload {
  stage: string;
  session_id: string;
}

export interface SSEStageCompletedPayload {
  stage: string;
  session_id: string;
  assistant_turn: AssistantTurn;
  progress: StageProgress;
  frontend: FrontendGuidance;
  sidebar: TriageSidebar;
}

export interface SSECompletePayload {
  response: TriageResponse;
}

export interface SSEErrorPayload {
  message: string;
  session_id: string;
}

File: Law_assistant/app/schemas/triage.py
from enum import Enum
from typing import Literal

from pydantic import BaseModel, Field


class TriageScenario(str, Enum):
    HOUSING = "housing"
    LABOR = "labor"
    CONSUMER = "consumer"
    DEBT = "debt"
    IP_REPUTATION = "ip_reputation"
    OTHER = "other"


class EvidenceSlotsModel(BaseModel):
    counterparty: str | None = None
    time_place: str | None = None
    amount: str | None = None
    agreement: str | None = None
    breach_fact: str | None = None
    existing_evidence: list[str] = Field(default_factory=list)


class SimilarCaseModel(BaseModel):
    title: str
    summary: str
    judgment: str
    takeaway: str


class TriageSidebar(BaseModel):
    evidence_slots: EvidenceSlotsModel
    action_steps: list[str] = Field(default_factory=list)
    demand_letter: str | None = None
    similar_cases: list[SimilarCaseModel] = Field(default_factory=list)


class ConversationMessageModel(BaseModel):
    role: Literal["user", "assistant"] = Field(description="消息角色")
    content: str = Field(description="消息内容")


class AssistantTurnModel(BaseModel):
    role: Literal["assistant"] = Field(default="assistant", description="助手消息角色")
    content: str = Field(description="本轮助手回复")
    stage: str = Field(description="本轮回复对应的图阶段")


class StageProgressModel(BaseModel):
    current_stage: str = Field(description="当前流程阶段")
    completed_stages: list[str] = Field(default_factory=list, description="已完成阶段")
    remaining_stages: list[str] = Field(default_factory=list, description="待完成阶段")
    next_stage: str | None = Field(default=None, description="下一阶段")
    collected_slots: list[str] = Field(default_factory=list, description="已补齐证据槽位")
    missing_slots: list[str] = Field(default_factory=list, description="缺失证据槽位")
    evidence_completion_ratio: float = Field(description="证据槽位完成度，范围 0 到 1")
    evidence_complete: bool = Field(description="证据是否已补齐")
    deliverables_ready: bool = Field(description="催告函和相似案例是否已可展示")


class FrontendGuidanceModel(BaseModel):
    display_mode: str = Field(description="前端当前建议展示模式")
    primary_panel: str = Field(description="当前优先展开的侧边栏区域")
    input_placeholder: str = Field(description="输入框占位提示")
    follow_up_questions: list[str] = Field(default_factory=list, description="建议追问列表")
    suggested_actions: list[str] = Field(default_factory=list, description="前端可展示的快捷动作")


class TriageRequest(BaseModel):
    user_input: str = Field(min_length=1, description="用户当前输入")
    scenario: TriageScenario | None = Field(
        default=None, description="纠纷场景；为空时默认路由到 other"
    )
    scenario_hint: TriageScenario | None = Field(
        default=None,
        description="场景参考提示；仅用于首轮真实案情输入时辅助路由",
    )
    session_id: str | None = Field(default=None, description="会话 ID")


class TriageResponse(BaseModel):
    session_id: str
    scenario: TriageScenario
    scenario_label: str
    assistant_message: str
    current_stage: str
    assistant_turn: AssistantTurnModel
    conversation_history: list[ConversationMessageModel] = Field(default_factory=list)
    progress: StageProgressModel
    frontend: FrontendGuidanceModel
    sidebar: TriageSidebar
